api/v1beta1/policy_types.go

```diff
@@ -91,6 +91,11 @@ type PolicySpec struct {
 	//
 	// Description holds the description string for the Role
 	Description string `json:"description,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	//
+	// AWSPolicyName is the name of the policy to create. If not specified, metadata.name will be used
+	AWSPolicyName string `json:"awsPolicyName,omitempty"`
```

Comment:
I wonder if this should simply be `name` rather than `awsPolicyName` as it's implied that it'll be an AWS Policy in any case.